### PR TITLE
Add deployment min/max options

### PR DESCRIPTION
**Pull request by beedub**
This will allow each process to specify the min and max number of
containers to run during deployment

### CODE REVIEW

File: api/models/manifest.go [hunk 1]
```diff
@@ -537,6 +537,23 @@ func (e ManifestEntry) MountableVolumes() []MountableVolume {
 	return volumes
 }
 
+// DeploymentMinimumPercent returns the min percent of containers that are allowed during deployment
+func (me ManifestEntry) DeploymentMinimumPercent() string {
+	if min := me.Label("convox.deployment.min_percent"); min != "" {
```
Comment (ddollar) [inline api/models/manifest.go]: Let's use `convox.deployment.minimum` for this

Comment (beedub) [inline api/models/manifest.go]: updated

**Comment by ddollar** [comment]
This looks good but similar concern as the others with removing the params. We should add a check to make sure that the labels match the old params if they were set.

File: api/models/release.go [hunk 2]
```diff
@@ -194,6 +194,19 @@ func (r *Release) Promote() error {
 			}
 		}
 
+		// Deployment min/max per service was moved from CF param to manifest label
+		// Verify that deployment min and max are backwards compatible
+		deploymentOpts := []string{ "minimum", "maximum" }
+		for opt := range deploymentOpts {
+			label := fmt.Sprintf("convox.deployment.%s", opt)
+			if labelVal, ok := entry.Labels[label]; ok {
+				cfVal, ok := app.Parameters[label]
```
Comment (beedub) [inline api/models/release.go]: @ddollar this should return the existing values in the template right?

Comment (ddollar) [inline api/models/release.go]: I think you want `app.Parameters["DeploymentMinimum"]` etc

Comment (beedub) [inline api/models/release.go]: you're right. updated

**Comment by ddollar** [comment]
Carrying forward in #974

**Pull request closed without merging**